Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCiscoFirepower.yaml

```diff
@@ -0,0 +1,248 @@
+          | where DeviceEventClassID has_any("INTRUSION:400", "PV:112", "RNA:1003:1")
+          | invoke _ASIM_ResolveNetworkProtocol('Protocol')
+          | extend NetworkProtocol = iff(NetworkProtocol == "Unassigned" and Protocol !in (63, 68, 99, 114, 253, 254), Protocol, NetworkProtocol);
+      let Connection_statistics_events = alllogs
+          | where DeviceEventClassID has "RNA:1003:1"
+          | parse-kv AdditionalExtensions as (
+              start: long,
+              end: long,
+              bytesIn: long,
+              bytesOut: long,
+              )
+              with (pair_delimiter=';', kv_delimiter='=')
+          | lookup EventResultDetailsLookup on Reason
+          | extend
+              SrcBytes = bytesIn,
+              DstBytes = bytesOut,
+              EventOriginalResultDetails = Reason,
+              AdditionalFields = bag_pack("policy", DeviceCustomString1,
+                                                  "ingressZone", DeviceCustomString3,
+                                                  "egressZone", DeviceCustomString4,
+                                                  "instanceID", ProcessID,
+                                                  "clientApplicationID", RequestClientApplication,
+                                                  "clientUrl", RequestURL);
+      let Intrusion_events = alllogs
+          | where DeviceEventClassID has "INTRUSION:400"
+          | parse-kv AdditionalExtensions as (
+              start: long
+              )
+              with (pair_delimiter=';', kv_delimiter='=')
+          | extend 
+              EventMessage = Activity,
+              ThreatCategory = DeviceEventCategory,
+              AdditionalFields = bag_pack("policy", DeviceCustomString1,
+                                                  "ingressZone", DeviceCustomString3,
+                                                  "egressZone", DeviceCustomString4,
+                                                  "ipspolicy", DeviceCustomString5,
+                                                  "clientApplicationID", RequestClientApplication,
+                                                  "clientUrl", RequestURL);
+      let Policy_Violation_events = alllogs
+          | where DeviceEventClassID has "PV:112"
+          | extend
+              EventMessage = Message,
+              AdditionalFields = bag_pack("policy", DeviceCustomString1,
+                                                  "ingressZone", DeviceCustomString3,
+                                                  "egressZone", DeviceCustomString4)
+          | project-rename DstUsername = DestinationUserName
+          | extend
+              DstUsernameType = _ASIM_GetUsernameType(DstUsername),
+              DstUserType = _ASIM_GetUserType(DstUsername, "");
+      let AllEvents = union isfuzzy=false Connection_statistics_events, Intrusion_events, Policy_Violation_events
```

Comment:
Done

Reply:
@jayeshprajapaticrest @niralishah-crest 
Please note, we should only use 'union' when it's absolutely necessary. It's an expensive function in terms of performance. In addition, in some parsers, we had to use 'Join' for threat which further impacts the performance. Please try to avoid union when possible.